common/scala/src/main/scala/whisk/connector/kafka/KafkaConsumerConnector.scala

```diff
@@ -89,9 +134,18 @@ class KafkaConsumerConnector(kafkahost: String,
     val keyDeserializer = new ByteArrayDeserializer
     val valueDeserializer = new ByteArrayDeserializer
     val consumer = new KafkaConsumer(props, keyDeserializer, valueDeserializer)
-    topics foreach { consumer.subscribe(_) }
+    topics.foreach(consumer.subscribe(_))
     consumer
   }
 
-  private val consumer = getConsumer(getProps, Some(List(topic)))
+  private def recreateConsumer(): Unit = {
+    val oldConsumer = consumer
+    Future {
+      oldConsumer.close()
+      logging.info(this, s"old consumer closed")
+    }
+    consumer = getConsumer(getProps, Some(List(topic)))
+  }
+
+  @volatile private var consumer = getConsumer(getProps, Some(List(topic)))
```

Comment:
i'm not sure that this needs to be volatile (and hence incur the performance implications); in the architecture (controller's pings and active acks, or invoker activation feed) there is one actor that is connected to the message consumer and it serializes all the reads from the message bus.